dds/idl/marshal_generator.cpp

```diff
@@ -1006,6 +1006,10 @@ namespace {
         be_global->impl_ <<
           "  CORBA::ULong length;\n"
           << streamAndCheck(">> length");
+        be_global->impl_ <<
+          "  if (length > strm.length()) {\n"
```

Comment:
`length` and `strm.length()` measure different things. `length` is the element count in the sequence and `strm.length()` is the number of bytes left in the stream.

Reply:
That's by design -- it's a check (like the one we already have in XTypes) to make sure a very large number of elements, more than are possible to have in the stream, aren't allocated

Reply:
Then there should be a comment, and maybe an error message, saying what's going on.